src/pages/Task.tsx

```diff
@@ -144,6 +180,45 @@ const TaskCore: React.FC = () => {
       />
       <PageLayout>
         <PageSider>
+          {task?.latestExecution > 0 && (
+            <StyledSelect
+              placeholder="Choose an execution"
+              disabled={executionsLoading}
+              key={execution}
+              data-test-id="execution-select"
+              value={`Execution ${ExecutionAsDisplay(
+                execution === null ? task?.latestExecution : execution
+              )}${
+                execution === null || execution === task?.latestExecution
+                  ? " (latest)"
+                  : ""
+              }`}
+              onChange={(selected: number | null) => {
+                setExecution(selected);
+                updateQueryParams({
+                  execution: `${ExecutionAsDisplay(selected)}`,
+                });
+              }}
+            >
+              {allExecutions?.map((singleExecution) => (
+                <Option
+                  key={singleExecution.execution}
+                  value={singleExecution.execution}
+                  data-test-id={`execution-${singleExecution.execution}`}
+                >
+                  <StyledSquare
+                    color={mapVariantTaskStatusToColor[singleExecution.status]}
+                  />
+                  <StyledP1>
+                    {" "}
+                    Execution {ExecutionAsDisplay(
+                      singleExecution.execution
+                    )} - {shortDate(singleExecution.createTime)}
+                  </StyledP1>
+                </Option>
+              ))}
+            </StyledSelect>
+          )}
```

Comment:
to avoid the messiness that adding this component to an existing component + all of its state and query hooks, it can be made into its own component that encapsulates the jsx and and all the state/query logic

Reply:
done